src/lib/services/transfer-service/TransferService.ts

```diff
@@ -1,27 +1,233 @@
 import { mnestixFetch } from 'lib/api/infrastructure';
-import { AssetAdministrationShell } from '@aas-core-works/aas-core3.0-typescript/dist/types/types';
-import { IAssetAdministrationShellRepositoryApi } from 'lib/api/basyx-v3/apiInterface';
-import { Submodel } from '@aas-core-works/aas-core3.0-typescript/types';
+import { IAssetAdministrationShellRepositoryApi, ISubmodelRepositoryApi } from 'lib/api/basyx-v3/apiInterface';
+import { ISubmodelRegistryServiceApiInterface } from 'lib/api/submodel-registry-service/ISubmodelRegistryServiceApiInterface';
+import { IRegistryServiceApi } from 'lib/api/registry-service-api/registryServiceApiInterface';
+import { IDiscoveryServiceApi } from 'lib/api/discovery-service-api/discoveryServiceApiInterface';
+import { RegistryServiceApi } from 'lib/api/registry-service-api/registryServiceApi';
+import { DiscoveryServiceApi } from 'lib/api/discovery-service-api/discoveryServiceApi';
+import { SubmodelRegistryServiceApi } from 'lib/api/submodel-registry-service/submodelRegistryServiceApi';
+import { AssetAdministrationShellDescriptor, Endpoint, SubmodelDescriptor } from 'lib/types/registryServiceTypes';
+import { TransferDto, TransferResult } from 'lib/services/transfer-service/transferActions';
+import { encodeBase64 } from 'lib/util/Base64Util';
+import { AssetAdministrationShell, Submodel } from '@aas-core-works/aas-core3.0-typescript/types';
 
 export class TransferService {
     private constructor(
         protected readonly targetAasRepositoryClient: IAssetAdministrationShellRepositoryApi,
-        protected readonly targetSubmodelRepositoryClient: SubmodelRepositoryApi,
+        protected readonly targetSubmodelRepositoryClient: ISubmodelRepositoryApi,
+        protected readonly targetAasDiscoveryClient?: IDiscoveryServiceApi,
+        protected readonly targetAasRegistryClient?: IRegistryServiceApi,
+        protected readonly targetSubmodelRegistryClient?: ISubmodelRegistryServiceApiInterface,
     ) {}
 
-    static create(targetAasRepositoryBaseUrl: string, targetSubmodelRepositoryBaseUrl: string): TransferService {
+    static create(
+        targetAasRepositoryBaseUrl: string,
+        targetSubmodelRepositoryBaseUrl: string,
+        targetAasDiscoveryBaseUrl?: string,
+        targetAasRegistryBaseUrl?: string,
+        targetSubmodelRegistryBaseUrl?: string,
+    ): TransferService {
         const targetAasRepositoryClient = AssetAdministrationShellRepositoryApi.create({
             basePath: targetAasRepositoryBaseUrl,
             fetch: mnestixFetch(),
         });
+
         const targetSubmodelRepositoryClient = SubmodelRepositoryApi.create({
             basePath: targetSubmodelRepositoryBaseUrl,
             fetch: mnestixFetch(),
         });
-        return new TransferService(targetAasRepositoryClient, targetSubmodelRepositoryClient);
+
+        const targetAasDiscoveryClient = targetAasDiscoveryBaseUrl
```

Comment:
So you already implemented the logic for the next story?

Reply:
We push to registry and discovery and have added the necessary method arguments for it, so yeah.

However, we expect it to be called either will null or the default services for now.
The next story would add the functionality to choose those services.